### PR TITLE
chore: get subsystem with nqn to list all subsystems for volume

### PR DESCRIPTION
**Pull request by abhilashshetty04**
This PR adds method to get all subsystems registered with the supplied nqn.

Signed-off-by: Abhilash Shetty <abhilash.shetty@datacore.com>

### CODE REVIEW

File: nvmeadm/src/nvmf_subsystem.rs [hunk 1]
```diff
@@ -172,6 +172,24 @@ impl Subsystem {
             Some(subsys) => Ok(subsys),
         }
     }
+
+    /// Gets all Nvme subsystems registered for a given volume.
+    pub fn get_from_nqn(nqn: &str) -> Result<Vec<Subsystem>, NvmeError> {
```
Comment (tiagolobocastro) [inline nvmeadm/src/nvmf_subsystem.rs]:
```suggestion
    pub fn try_from_nqn(nqn: &str) -> Result<Vec<Subsystem>, NvmeError> {
```

File: nvmeadm/src/nvmf_subsystem.rs [hunk 2]
```diff
@@ -172,6 +172,24 @@ impl Subsystem {
             Some(subsys) => Ok(subsys),
         }
     }
+
+    /// Gets all Nvme subsystems registered for a given volume.
```
Comment (tiagolobocastro) [inline nvmeadm/src/nvmf_subsystem.rs]:
```suggestion
    /// Gets all Nvme subsystems registered for a given nqn.
```

File: nvmeadm/src/nvmf_subsystem.rs [hunk 3]
```diff
@@ -172,6 +172,24 @@ impl Subsystem {
...
+    pub fn get_from_nqn(nqn: &str) -> Result<Vec<Subsystem>, NvmeError> {
+        let nvme_subsystems = NvmeSubsystems::new()?;
+        let mut nvme_paths = vec![];
+        for path in nvme_subsystems.flatten() {
+            if path.nqn == nqn && (path.state == "live" || path.state == "connecting") {
```
Comment (tiagolobocastro) [inline nvmeadm/src/nvmf_subsystem.rs]: Should we split this into 2 fns, one that gets matching non subsystems, and another than filters it down to only live or connecting paths?

File: nvmeadm/src/nvmf_subsystem.rs [hunk 4]
```diff
@@ -172,6 +172,24 @@ impl Subsystem {
...
+                nvme_paths.push(path);
+            }
+        }
+        if !nvme_paths.is_empty() {
```
Comment (tiagolobocastro) [inline nvmeadm/src/nvmf_subsystem.rs]: isn't this logic reversed?

File: nvmeadm/src/error.rs [hunk 5]
```diff
@@ -34,6 +34,8 @@ pub enum NvmeError {
         host: String,
         port: u16,
     },
+    #[snafu(display("No Nvmf Subsystem found for nqn :{}", nqn))]
+    NoSubsytemFound { nqn: String },
```
Comment (tiagolobocastro) [inline nvmeadm/src/error.rs]: We already have an `NqnNotFound` error, could we reuse it?

File: nvmeadm/src/nvmf_subsystem.rs (hunk 3 above)
Comment (tiagolobocastro) [reply nvmeadm/src/nvmf_subsystem.rs]: Maybe the filtering for live or connecting can be done on the control-plane repo

File: nvmeadm/src/error.rs (hunk 5 above)
Comment (abhilashshetty04) [reply nvmeadm/src/error.rs]: I thought that error is not descriptive enough. Made minor change in formating and using it.

File: nvmeadm/src/nvmf_subsystem.rs (hunk 3 above)
Comment (abhilashshetty04) [reply nvmeadm/src/nvmf_subsystem.rs]: Incorporated

File: nvmeadm/src/nvmf_subsystem.rs (hunk 4 above)
Comment (abhilashshetty04) [reply nvmeadm/src/nvmf_subsystem.rs]: Yes it was wrong. Fixed it

File: nvmeadm/src/error.rs (hunk 5 above)
Comment (abhilashshetty04) [reply nvmeadm/src/error.rs]: Incorporated

Comment (tiagolobocastro) [reply nvmeadm/src/error.rs]: Then you can fix that error and make it more descriptive, win win :)

Comment (tiagolobocastro) [reply nvmeadm/src/error.rs]: Oh I see that's what you did already, great! 👍

File: nvmeadm/src/nvmf_subsystem.rs (hunk 2 above)
Comment (abhilashshetty04) [reply nvmeadm/src/nvmf_subsystem.rs]: Incorporated

**Review by tiagolobocastro — APPROVED** [review]
LGTM

**Pull request merged**